airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/steps/gradle.py

```diff
@@ -32,210 +36,129 @@ class GradleTask(Step, ABC):
+            yield
+        except Exception as e:
+            raise InvalidGradleEnvironment(e)
+        finally:
+            if self.context.s3_build_cache_access_key_id and self.context.s3_build_cache_secret_key:
+                del os.environ["S3_BUILD_CACHE_ACCESS_KEY_ID"]
+                del os.environ["S3_BUILD_CACHE_SECRET_KEY"]
+            # Remove secrets from the secrets folders only in CI
+            if self.context.is_ci:
+                for secret_path in secrets_written:
+                    secret_path.unlink()
+
+    def _run_gradle_in_subprocess(self) -> Tuple[str, str, int]:
+        """
+        Run a gradle command in a subprocess.
+        """
+        try:
+            self.context.logger.info(f"Running gradle command: {self.gradle_command}")
+            process = subprocess.Popen(
+                self.gradle_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, universal_newlines=True
             )
-            # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
-            # This is consumed by settings.gradle in the repo root.
-            .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
-            # Disable the Ryuk container because it needs privileged docker access which it can't have.
-            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true")
-            # Set the current working directory.
-            .with_workdir("/airbyte")
-        )
 
-        # Augment the base container with S3 build cache secrets when available.
-        if self.context.s3_build_cache_access_key_id:
-            gradle_container_base = gradle_container_base.with_secret_variable(
-                "S3_BUILD_CACHE_ACCESS_KEY_ID", self.context.s3_build_cache_access_key_id.as_dagger_secret(self.dagger_client)
-            )
-            if self.context.s3_build_cache_secret_key:
-                gradle_container_base = gradle_container_base.with_secret_variable(
-                    "S3_BUILD_CACHE_SECRET_KEY", self.context.s3_build_cache_secret_key.as_dagger_secret(self.dagger_client)
-                )
+            stdout, stderr = process.communicate()
+
+            if process.returncode != 0:
+                stderr = f"Error while running gradle command: {self.gradle_command}\n{stderr}"
+                return stdout, stderr, process.returncode
 
-        # Running a gradle task like "help" with these arguments will trigger updating all dependencies.
-        # When the cache is cold, this downloads many gigabytes of jars and poms from all over the internet.
-        warm_dependency_cache_args = ["--write-verification-metadata", "sha256", "--dry-run"]
-        if self.context.is_local:
-            # When running locally, this dependency update is slower and less useful than within a CI runner. Skip it.
-            warm_dependency_cache_args = ["--dry-run"]
+            return stdout, stderr, process.returncode
+        finally:
```

Comment:
if the started process hangs, how does this finally block trigger? 

this is probably my lack of python knowledge

Reply:
If it hangs it's never reached. Good point. I can add a timeout.